google/cloud/bigquery/job/base.py

```diff
@@ -659,6 +659,12 @@ def __init__(self, job_type, **kwargs):
         for prop, val in kwargs.items():
             setattr(self, prop, val)
 
+    def __setattr__(self, name, value):
+        """Override to be able to raise error if an unknown property is being set"""
+        if not name.startswith("_") and name not in type(self).__dict__:
```

Comment:
Let's try using [hasattr](https://docs.python.org/3/library/functions.html#hasattr) to see if that accounts for superclass properties.

```suggestion
        if not name.startswith("_") and not hasattr(self, name):
```

Reply:
Ok, but we have to use type(self), because if not the properties are looked up on the instance.